app/src/main/java/com/example/testcompras/iu/view/MainActivity.kt

```diff
@@ -0,0 +1,42 @@
+package com.example.testcompras.iu.view
+
+import android.os.Bundle
+import androidx.activity.enableEdgeToEdge
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.view.ViewCompat
+import androidx.core.view.WindowInsetsCompat
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProvider
+import com.example.testcompras.R
+import com.example.testcompras.data.DataRepository
+import com.example.testcompras.databinding.ActivityMainBinding
+import com.example.testcompras.domain.GetDataUseCase
+import com.example.testcompras.iu.viewModel.MainViewModel
+import com.example.testcompras.iu.viewModel.modelFactory.MainViewModelFactory
+
+class MainActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityMainBinding
+    private lateinit var viewModel: MainViewModel
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        enableEdgeToEdge()
+        setContentView(binding.root)
+        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
+            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
+            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
+            insets
+        }
+        val repository = DataRepository()
+        val getDefaultTextUseCase = GetDataUseCase(repository)
+
+        val factory = MainViewModelFactory(getDefaultTextUseCase)
+        viewModel = ViewModelProvider(this, factory)[MainViewModel::class.java]
```

Comment:
Why do you use a factory instead of a normal constructor?

`MainViewModel(getDefaultTextUseCase)`